# Final Analysis_2 

## 1. Problem statement (summarized)

- How are the negative reviews distribution look like, given we have already identify groups of sellers to focus on.


Our key findings so far :
- The `wait_time` was the most significant factor behind low review scores, but reading comments of the bad reviews also showed that some of them were linked to the seller or to the product itself.
- `wait_time` is made up of seller's `delay_to_carrier` + `carrier_delivery_time`. The latter being outside of Olist's direct control, improving it is not a quick-win recommendation we can make to Olist CEO without in-depth analysis of their operational practices.
- On the contrary, a better selection of `sellers` can positively impact the `delay_to_carrier` and reduce the number of bad `review_scores` on Olist.

In [1]:
from olist.data import Olist

import translators as ts
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

Using state Perak server backend.


In [34]:
data = Olist().get_data()
reviews = data["order_reviews"]

one_star_reviews = reviews[(reviews["review_score"] == 1) & (~reviews["review_comment_message"].isna())]
one_star_reviews.loc[:,'review_comment_message'] = one_star_reviews['review_comment_message'].astype('str')
print(one_star_reviews.shape)
one_star_reviews.head()

(8745, 7)


/home/ysin/.pyenv/versions/3.8.12/envs/lewagon/lib/python3.8/site-packages/pandas/core/indexing.py:1773: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(ilocs[0], value, pi)


review_id                          order_id  \
19  373cbeecea8286a2b66c97b1b157ec46  583174fbe37d3d5f0d6661be3aad1786   
29  2c5e27fc178bde7ac173c9c62c31b070  0ce9a24111d850192a933fcaab6fbad3   
32  58044bca115705a48fe0e00a21390c54  68e55ca79d04a79f20d4bfc0146f4b66   
39  9fd59cd04b42f600df9f25e54082a8d1  3c314f50bc654f3c4e317b055681dff9   
51  e233e51d11511bf30e568c76360ace52  548df2c6e5f089574614894bca78acf5   

    review_score     review_comment_title  \
19             1  Não chegou meu produto    
29             1                      NaN   
32             1                      NaN   
39             1                      NaN   
51             1                      NaN   

                               review_comment_message review_creation_date  \
19                                            Péssimo  2018-08-15 00:00:00   
29                Não gostei ! Comprei gato por lebre  2017-12-13 00:00:00   
32  Sempre compro pela Internet e a entrega ocorre...  2018-04-08 00:00:00   
39                       Nada de chegar o meu pedido.  2017-04-21 00:00:00   
51  recebi somente 1 controle Midea Split ESTILO.\...  2018-01-28 00:00:00   

   review_answer_timestamp  
19     2018-08-15 04:10:37  
29     2017-12-16 07:14:07  
32     2018-04-09 12:22:39  
39     2017-04-23 05:37:03  
51     2018-02-04 11:06:15

In [35]:
def translation(data,series):
    name = "translated_"+series.name.split("_")[-1]
    
    for i in range(series.shape[0]):
        data.loc[i,name]= ts.google(data.loc[i,series.name], from_language = "pt", to_language = "en")

In [25]:
%%time
resampled_reviews = one_star_reviews.sample(50).reset_index()
translation(resampled_reviews, resampled_reviews['review_comment_message'])
resampled_reviews

CPU times: user 1.14 s, sys: 80.6 ms, total: 1.22 s
Wall time: 1min 24s


index                         review_id                          order_id  \
0   50133  2521ded73db8561e0fd8df85eeeaaf5e  265271dd979e1221bab55d039af24b37   
1   90102  afbdacbc3c4264261766988251038ef3  72e4ee7d458b36222813e2b9707d9d25   
2   11047  f495b89455116ac326d54a32ee939a00  2dee1b58f6f3c724427ad545adf6d5e4   
3   38740  f763252b810a719d655523cd5b748c37  e43508515d5e7a615643e552dc54a970   
4   42099  2af2f4bb459f8df919aaaadbe64e7cad  a9e3336634f202f7806561c5f41b2e1a   
5   66728  a765ade5a94437b1d9dd9969f30149f2  e47c4c77d188f893a6534dbca93d1ef0   
6   11896  d0517d82a887cfa4d60bbd1790ec563b  5f3ce710839dcf62389d5187ece2d4e8   
7   94004  b7013a9e63f944b85e8e44ee177f0b0c  6adb948b069c269ba13ff1c38ad90228   
8   37848  55bc097bf380cffd36a1d0a9efe200ff  e3db241d7377ac91f1253454bdbeafe4   
9   27107  dd3e09a802675667fb9761109edea549  e95f655fc4bb105e8915ab28d7bd0ec6   
10  90132  326bf102dfd70ae9f5a1f8cbd440cc2e  4fc7fe1bc9c28e87c8d77a28aefb97f4   
11  15025  510eaabfb5a0b383d15b36c17ce1e1b4  af4b4e5e7cc290c42e0935df9ff3479c   
12  59027  5fc891a01bff62b873bcf6cf2e2375bb  d2feb2f430d7d97b41873672953fe944   
13  70789  d2dec0f33eb8b8ce57bf756953b45f3d  72169ffd15efc268ce012f931a368c4e   
14  59048  3fcdc59670b97dcbf747e0a9d6888fd4  a47613180529472d74824b41a0564e3f   
15  54425  99241e07e2f135e5aadbf34f46398b27  9bcddf79e9ba13cbf6c32e0916bd42d7   
16   8342  0f7dda01ef285792786f337afc122956  b85f9848a67767d36a4c39d1237b471a   
17  66148  7a1333349414c44f06bbfd875af2e672  bf488c4d02bcdb68c278b42d9417f05d   
18  58809  5dd59df9921255e70037d6ec56496fc6  25348d7abfa3893731e3836aab2f6a6c   
19  95891  4a4cc14b3fd268babf164f104c7fa928  8bde999dc4ad8ff2e39ca01c02a81119   
20   3524  4048b81e0de31798b68dcdb9bdf63bb0  333977327b5c548879a21a3b1fc87c48   
21  98372  1737d7268e9e82102dd511df4e3bdf02  8b6ed0131b6379b73b0dc313d55eb377   
22  75350  c099051f5050d2408fbe5d33beb8a8f8  208f886949aabb97ed60c2f51a74717b   
23  34670  08b0990850e58e9b24a57ba4e593f5db  cfa5f05fcf442b5faa1cffd73b4fd9ee   
24  88878  2cdd1f60be27bdfbffa32251283edd2a  879a1b657dd0ac822411797d57cfe29f   
25   1582  985e48b93ee65708e73e0d29b1bde5ea  43101ee6b066954af2c1074445d240ec   
26  82911  5907db64dd9ab47d9058b074db27457b  ff14a3e36b78f407a4e8402572e4edfe   
27  63589  eeff034df4f8096ba692c63d6945946a  873c916c2bff3ea61a59f7bf7f0879c2   
28  63199  88017f91d6bdee7c0eead9b64529323a  cccb57e7a01fe8d245ce2027974b455e   
29  25846  d1950ebb57352ea39897dbb1817caa93  36272970346cb4aadae6ef0defd79773   
30  73369  a74aaa15cf2143dfb74a11fe27ac8d77  e8cab458223cfed14e73ea0d7fe1a511   
31  24780  3fe4254e4bd3f7a0cc099775a7f064b0  1b0ed67e2931f09f00cd8553479f5823   
32  84591  9b20657e2aa298e62ef5bbcbd29dd445  57c28fab977b1d61c2b5612548fda073   
33  98881  9b4c6ceccd5fa257e75239739722c336  ccbf44556216d2ad81b039514a0d8877   
34  27951  d5c5e1e70f31a5a321ab3acc92306f80  5f2aaf089bdf0d14f2357066f139c2a4   
35  31135  59800f125055137c475661cd350c9def  56301ae69daec31f44fa405fc3a09518   
36  96687  c0881b2797529c9af9325018e7fbb694  fd7eb2117233aa5ac29e94a28e082077   
37   7168  93f529187a4579f3361f118e2ad2515d  f1c5f3e9df9be5699de80cb4eec88c04   
38  69866  10d29e257f170386e5a0808100c2e525  afd363f0f987b7f9bf8e2e1b3cb543dd   
39  82441  dfe24a6167ea9ecd3ab623ac6ca76add  82f71912ce7c7eab50c463a9bc595af6   
40  80170  40948f7f28945e16a89a912386666c7c  94b860984e79a390fa2d4793ddd46404   
41  81385  383c30fe15b6d0484c339ad1d3c4d055  a4887ea2ccd5e37945934525a9939e9c   
42  49527  1cf67d4ae88f6326fc65abb053213edc  c6a7876557e3e3e31f229852dfa4f5c3   
43  50607  c6c4823bd73f5e812a1d29f6abd0ebf5  6434ca73491ebf0b471702a2506eb871   
44  16624  ea0bd223587de0d11dda627671dac08c  e2557098c56afd06a929b32a63b5f856   
45  16195  c08738e6a4f03cdbcad1c12bd1233339  c21fcb36121a9b1c65eed7b20f0862bd   
46  86172  4a317fd181969db85d0cd7353d17150f  bf80d30d4654a34a647d0e5db84a5e2e   
47  24734  606dad75a0a047aa73ecc358e3f59642  2711089c7fec59d4dc8483e3c6a12fa3   
48  86080  eb85bd52d6e9a29cc3021c64

Review title is not used as most of them are empty. As seen above, we sampled 50 reviews and translated their comment message using google translator package. Translation took roughly 1.3 minutes for 50 reviews. Due to the time constraint (all reviews - 8745 of them will need around 4 hours to translate), we will only sample around 500 reviews to for analysis.